Docs: Edit 13.0.0 docs

kirszenbaum opened the pull request:
This PR:
- Adds a new migration guide: [Migrate from 12.4 to 13.0](http://localhost:8080/docs/javascript-data-grid/migration-from-12.4-to-13.0/)
- Edits the 13.0 changelog entries
- Adds 13.0 release notes
- Makes other minor changes

[skip changelog]

kirszenbaum commented:
@jansiegel – mind that [here](https://github.com/handsontable/handsontable/pull/10418/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR26) I changed the Angular version do 13 and lower. That's because Angular [no longer supports v12](https://angular.io/guide/releases#actively-supported-versions) and, , according to our new frameworks support policy, we don't support it either.

On the other hand, the new CodeSandbox Angular demo uses v12: https://codesandbox.io/s/handsontable-angular-data-grid-hello-world-app-13-0-0-49m644

jansiegel commented:
> no longer supports v12

We _could_ say something along the lines of: "Although we don't support angular<14, the wrapper is currently built using angular@12, so it works with angular versions 12+, but it may change without notice with any non-major release.", but I don't know if we _want_ to.
I, personally, refactored the wrapper to use angular@12 because I wanted it to support as many angular versions as possible.

@adamu-handsontable, could you help us out here?

kirszenbaum commented:
> no longer supports v12

@jansiegel – I updated this bit, as discussed [here](https://handsoncode.slack.com/archives/CCAJN0V1V/p1687419244577019?thread_ts=1687373215.799909&cid=CCAJN0V1V).

> - **Breaking change (Angular)**: Dropped support for Angular 13 and lower. From now on, Handsontable supports only those versions of Angular that are officially supported by the Angular team: currently, it's 14-16. However, Handsontable 13.0.0 was thoroughly tested and, to the best of our knowledge, works correctly with versions down to Angular 12. [#10396](https://github.com/handsontable/handsontable/pull/10396)

jansiegel commented on docs/content/guides/upgrade-and-migration/migrating-from-12.4-to-13.0.md:
```diff
@@ -0,0 +1,123 @@
+
+## Update to Angular 12 or higher
+
+Handsontable 13.0 requires Angular 12 or higher. If you're using an older version of Angular, you need to update it.
```
@kirszenbaum This is probably worth rephrasing as well.

kirszenbaum replied:
Ok, done
